neo/Network/P2P/Helper.cs

```diff
@@ -1,10 +1,80 @@
 ﻿using Neo.Network.P2P.Payloads;
+using System;
 using System.IO;
+using System.IO.Compression;
+using K4os.Compression.LZ4;
+using K4os.Compression.LZ4.Streams;
+using Neo.Network.P2P.Payloads;
 
 namespace Neo.Network.P2P
 {
     public static class Helper
     {
+        private static readonly LZ4EncoderSettings CompressSettings = new LZ4EncoderSettings() { CompressionLevel = LZ4Level.L04_HC };
+
+        private static readonly LZ4DecoderSettings DecompressSettings = new LZ4DecoderSettings() { };
+
+        public static byte[] DecompressGzip(this byte[] data)
+        {
+            using (var output = new MemoryStream())
+            using (var input = new MemoryStream(data))
+            using (var decoder = new GZipStream(input, CompressionMode.Decompress))
+            {
+                int nRead;
+                byte[] buffer = new byte[1024];
+
+                while ((nRead = decoder.Read(buffer, 0, buffer.Length)) > 0)
+                {
+                    output.Write(buffer, 0, nRead);
+                }
+
+                return output.ToArray();
+            }
+        }
+
+        public static byte[] DecompressLz4(this byte[] data)
+        {
+            using (var output = new MemoryStream())
+            using (var input = new MemoryStream(data))
+            using (var decoder = LZ4Stream.Decode(input, DecompressSettings, true))
+            {
+                int nRead;
+                byte[] buffer = new byte[1024];
+
+                while ((nRead = decoder.Read(buffer, 0, buffer.Length)) > 0)
+                {
+                    output.Write(buffer, 0, nRead);
+                }
+
+                return output.ToArray();
+            }
+        }
+
+        public static byte[] CompressLz4(this byte[] data)
+        {
+            using (var stream = new MemoryStream())
+            using (var encoder = LZ4Stream.Encode(stream, CompressSettings, true))
```

Comment:
Why not use Blocks but use Streams?

Reply:
Is better? we can use it if has benefits

Reply:
Stream compression has magic number, but generates bigger data.

Reply:
I created a PR for using blocks compression: https://github.com/shargon/neo/pull/10

Let's see which one is better.

Reply:
https://github.com/lz4/lz4/issues/428#issuecomment-350168943